docs/jax-101/05.1-pytrees.md

```diff
@@ -255,18 +279,20 @@ def unflatten_MyContainer(
   """Converts aux data and the flat contents into a MyContainer."""
   return MyContainer(aux_data, *flat_contents)
 
-jax.tree_util.register_pytree_node(
-    MyContainer, flatten_MyContainer, unflatten_MyContainer)
+jax.tree_util.register_pytree_with_keys(
+    MyContainer, flatten_with_keys_MyContainer, unflatten_MyContainer)
 
 jax.tree_util.tree_leaves([
     MyContainer('Alice', 1, 2, 3),
     MyContainer('Bob', 4, 5, 6)
 ])
 ```
 
+Note that registering both keys and values is a new feature, and you may see legacy pytrees that only has values registered (aka. registered via `register_pytree_node`). They are free to use most `tree_util` utilities except the `.*_with_path` APIs.
```

Comment:
> They are free to use most `tree_util` utilities except the `.*_with_path` APIs.

IIUC we have a decision to make here: do we make the _with_path APIs error when applied to pytrees with node types that don't opt into the extended API, or do we just handle those generically using flattened indices (i.e. the `FlattenedIndexKey` stuff)? If the latter, which I think is the current behavior of the code, we should change the second sentence not to imply that they're incompatible but just say they get a generic fallback.

Reply:
Thanks for catching that! We do allow the fallback of _with_path API on non-keypath node types. I added a showcase paragraph.